main.py

```diff
@@ -33,14 +34,18 @@ def __init__(self, input_lines: List[str]):
         self.result = ParsingResult()
         self.complete_parse()
 
+        self.versions_parser = VersionsParser(self.lines)
+
     def complete_parse(self):
-        # todo: :)
+        self.versions_parser.complete_parse()

```

Comment:
Have a look at this line, some test fails on it. I have also tried to run it and got an exception "'ParseDocument' object has no attribute 'versions_parser'"